pkg/controller/migmigration/backup.go

```diff
@@ -358,6 +443,93 @@ func (t *Task) annotateStorageResources() error {
 		}
 	}
 
+	return nil, resticPodCount
+}
+
+func (t *Task) areStagePodsCreated() (bool, error) {
+	client, err := t.getSourceClient()
+	if err != nil {
+		return false, err
+	}
+	// check if we have already created dummy pods
+	// if so, return true
+	uniqueBackupLabelKey := fmt.Sprintf("%s-%s", pvBackupLabelKey, t.PlanResources.MigPlan.UID)
+	labelSelector := map[string]string{
+		fmt.Sprintf("%s-copy", uniqueBackupLabelKey): "true",
+	}
+	options := k8sclient.MatchingLabels(labelSelector)
+	podList := corev1.PodList{}
+	err = client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return false, err
+	}
+	podCount := len(podList.Items)
+	if podCount == 0 {
+		return false, nil
+	}
+	readyCount := 0
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodRunning {
+			readyCount += 1
+		}
+	}
+	// Check if all pods are ready
+	if readyCount == podCount {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (t *Task) createStagePods() error {
+	client, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+
+	// Dummy pods haven't been created yet, lets create them
+	uniqueBackupLabelKey := fmt.Sprintf("%s-%s", pvBackupLabelKey, t.PlanResources.MigPlan.UID)
+	labelSelector := map[string]string{
+		uniqueBackupLabelKey: pvBackupLabelValue,
+	}
+	options := k8sclient.MatchingLabels(labelSelector)
+	podList := corev1.PodList{}
+	err = client.List(context.TODO(), options, &podList)
+	if err != nil {
+		return err
+	}
+	for _, pod := range podList.Items {
+		if pod.Annotations[resticPvBackupAnnotationKey] == "" {
+			continue
+		}
+		dummyPod := corev1.Pod{}
+		dummyPod.Spec.Containers = pod.Spec.Containers
+		dummyPod.Spec.Volumes = pod.Spec.Volumes
+		// Swap image ref, command, and args and create a new dummy pod to be backed up
+		for i, _ := range dummyPod.Spec.Containers {
```

Comment:
Has this been fixed up to the point where it no longer copies over unneeded (breaking) info from other the original Pod definitions?

Reply:
Not yet, doing that today.